plugin/trino-mongodb/src/test/java/io/trino/plugin/mongodb/TestMongoConnectorTest.java

```diff
@@ -149,6 +150,34 @@ public void testSortItemsReflectedInExplain()
                 "TopNPartial\\[count = 5, orderBy = \\[nationkey DESC");
     }
 
+    @Test
+    void testNonLowercaseCollection()
+    {
+        String schema = "test_db_" + randomNameSuffix();
+        String table = "test_collection_" + randomNameSuffix();
+        try {
+            MongoDatabase db = client.getDatabase(schema);
+            db.createCollection(table);
+            db.getCollection(table).insertOne(new Document("lowercase", 1));
+
+            assertThatThrownBy(() -> client.getDatabase(schema.toUpperCase(ENGLISH)).createCollection(table))
+                    .hasMessageContaining("db already exists with different case");
+
+            db.createCollection(table.toUpperCase(ENGLISH));
+            db.getCollection(table.toUpperCase(ENGLISH)).insertOne(new Document("uppwercase", 2));
+
+            assertThat(query("SELECT * FROM information_schema.tables WHERE table_catalog = 'mongodb' AND table_schema = '" + schema + "'"))
+                    .matches("VALUES (VARCHAR 'mongodb', VARCHAR '" + schema + "', VARCHAR '" + table + "', VARCHAR 'BASE TABLE')");
+            assertThat(query("SELECT table_name, column_name FROM information_schema.columns WHERE table_catalog = 'mongodb' AND table_schema = '" + schema + "'"))
+                    .matches("VALUES (VARCHAR '" + table + "', VARCHAR 'lowercase')");
```

Comment:
The first created object would have the precedence right ? Can we try with different case like camelcase as well.